lib/impl/AwsS3Tvm.js

```diff
@@ -109,9 +111,47 @@ class AwsS3Tvm extends Tvm {
       secretAccessKey: stsRes.Credentials.SecretAccessKey,
       sessionToken: stsRes.Credentials.SessionToken,
       expiration: stsRes.Credentials.Expiration,
-      params: { Bucket: params.s3Bucket }
+      params: { Bucket: bucketName }
     }
   }
 }
 
+function checkAndCreateS3Bucket(namespace, prefix, region) {
+  return new Promise((resolve, reject) => {
+    var s3 = new AWS.S3({region: region});
+    const bucketName = getBucketName(namespace, prefix)
+    var params = {
+    Bucket: bucketName
+    };
+    //check if bucket exists
+    s3.headBucket(params, function(err, data) {
+     if (err){
+       if(err.statusCode == 404 ) {
+         console.log("Trying to create bucket - " + bucketName)
+         s3.createBucket(params, function(err, data) {
+           if (err) {
+             console.log("Error creating bucket - "+err)
+             reject(err)
+           }
+           else {
+             console.log("Bucket created - " + bucketName)
+             resolve(bucketName)
+           }
+         });
+       }
+       else
+        reject(err)
+     }
+     else{
+       console.log("Found bucket - "+bucketName);
+       resolve(bucketName)
+     }
+    });
+  });
+}
+
```

Comment:
after creating the bucket let's add a tag  to be able to know to which namespace it belongs (billing, debugging,...)
```javascript 
const params= {
  Bucket: bucketName, 
  Tagging: { TagSet: [ { Key: "aio-rt-namespace", Value: namespace }] }
 }
await s3.putBucketTagging(params).promise()
```

Reply:
(or just namespace as key is fine aswell)